### PR TITLE
Search parameters in the URL

### PR DESCRIPTION
**Pull request by sandikbarr**
Adds search and paging parameters to the URL as defined in issue #55

### CODE REVIEW

File: src/app/search/search.component.ts [hunk 1]
```diff
@@ -20,152 +22,105 @@ export class SearchComponent implements OnInit {
...
-  search(page: number, per: number = undefined) {
-    if (this.isOnSeriesTab()) {
-      this.searchSeries(page, per);
-    } else if (this.isOnStoriesTab) {
-      this.searchStories(page, per);
+  navigateToPagePer(page: number, perPage: number = undefined) {
+    if (this.selectedTab === SearchComponent.TAB_STORIES) {
+      this.searchWithParams(Object.assign({}, this.searchStoryParams, {tab: this.selectedTab, page, perPage}));
+    } else if (this.selectedTab === SearchComponent.TAB_SERIES) {
+      this.searchWithParams(Object.assign({}, this.searchSeriesParams, {tab: this.selectedTab, page, perPage}));
```
Comment (sandikbarr) [inline src/app/search/search.component.ts]: I was worried that I might need a polyfill for Object (we did on a Babel project) but typescript takes care of it. I miss having the spread operator for objects though, which I have a feeling will make me sad when doing Redux.

File: src/app/search/search.component.ts [hunk 2]
```diff
@@ -20,152 +22,105 @@ export class SearchComponent implements OnInit {
...
+      this.auth = auth;
+
+      this.allSeriesIds = [-1];
+      this.allSeries = {};
+      this.auth.followItems('prx:series', {filters: 'v4'}).subscribe((series) => {
+        for (let s of series) {
+          this.allSeriesIds.push(s.id);
+          this.allSeries[s.id] = new SeriesModel(this.auth, s, false);
         }
+        this.subscribeRouteParams();
```
Comment (cavis) [inline src/app/search/search.component.ts]: Why wait for the CMS requests to parse the route params?

File: src/app/search/directives/search-series-form.component.css [hunk 3]
```diff
@@ -1,3 +1,7 @@
+form {
+  display: block;
```
Comment (sandikbarr) [inline src/app/search/directives/search-series-form.component.css]: No, forms.css makes them display:flex. I haven't verified that's actually being used though. This file needs to go away eventually though as it's very generic styling that belongs in the styleguide.

File: src/app/search/search.component.ts (hunk 2 above)
Comment (sandikbarr) [inline src/app/search/search.component.ts]: The Observable on the route params parses the route params then calls the search functions which need the auth and/or series to make requests on.

Comment (sandikbarr) [inline src/app/search/search.component.ts]: sounds like we need to enter an issue for this!

**Pull request merged**